stl/inc/cmath

```diff
@@ -620,8 +620,7 @@ _STD _Common_float_type_t<_Ty1, _Ty2> remquo(_Ty1 _Left, _Ty2 _Right, int* _Pquo
     template <class _Ty1, class _Ty2,                                                                            \
         _STD enable_if_t<_STD is_arithmetic_v<_Ty1> && _STD is_arithmetic_v<_Ty2>, int> = 0>                     \
     _NODISCARD _STD _Common_float_type_t<_Ty1, _Ty2> NAME(_Ty1 _Left, _Ty2 _Right) noexcept /* strengthened */ { \
-        using _Common = _STD _Common_float_type_t<_Ty1, _Ty2>;                                                   \
-        return FUN(static_cast<_Common>(_Left), static_cast<_Common>(_Right));                                   \
+        return FUN(static_cast<double>(_Left), static_cast<double>(_Right));                                     \

```

Comment:
won't this induce narrowing warnings if common float type is "float"? (since it'll still return float).

Also, if the common type is "float", couldn't this reduce the accuracy of the result because of the additional rounding?

Reply:
oh, the cmath functions are already returning double in all cases. Why did this not give narrowing warnings in the old code?

Reply:
> Why did this not give narrowing warnings in the old code?

The only way for the common type to be `float` is if both arguments are `float`, in which case (I think) overload resolution will prefer the non-template global `float copysign(float,float)` overload (via `namespace std { using _CSTD copysign; }`).

Reply:
`_Common_float_type_t<T, U>` is `float` only when both `T` and `U` are `float`. When `T` and `U` are both `float` (or both `long double` , for that matter), overload resolution prefers the non-template `meow` overloads to the `meow` templates generated from this macro. The template is only selected when both arguments have integral type or the types of the two arguments differ. This crucial non-obvious fact deserves a comment.